cross-validation/boosted-ensemble/script.whizzml

```diff
@@ -172,6 +195,29 @@
   (dataset ["fields" objective-id "name"] false))
 
 
+;; Stratified sampling: Keeping proportions in sampling across classes
+
+(define (stratified-k-folds dataset k-folds objective-id)
+  (let (summary (dataset ["fields" objective-id "summary" "categories"] [])
+        cat-names (map (lambda (x) (x 0)) summary)
+        cat-ds-ids (for (cat cat-names)
+                     (create-k-folds
+                       (create-dataset {"origin_dataset" (dataset "resource")
+                                        "lisp_filter" (flatline
+                                                        "(= "
+                                                        "(f {objective-id})"
+                                                        " {{cat}})")
+                                        "name" (str (dataset "name")
+                                                    " - "
+                                                    cat)})
+                       k-folds))
+        ds-ids (for (index (range 0 (count (cat-ds-ids 0))))
+                 (create-dataset {"origin_datasets" (map (lambda (x) (x index))
+                                                         cat-ds-ids)})))
```

Comment:
if i'm reading this correctly, you want to map over the transposed
`cats-ds-ids`, right? if so, a shorter way of transposing `m`
if `(apply map list m)`

```
ds-ids (map (lambda (ids) (create-dataset {"origin_datasets" ids}))
            (apply map list cats-ds-ids))
```

we could have a `transpose` defined in he stdlib... if you use this
trick here, i would define a little `_transpose` function to make
it clearer.